src/materials/ConcreteThermalMoisture.C

```diff
@@ -382,206 +400,212 @@ ConcreteThermalMoisture::computeProperties()
-    if (_eqv_age[qp] >= 5)
-      N_te = 2.5 + 15.0 / _eqv_age[qp];
-
-    Real n = N_te * N_wc * N_ct * 1.0;
-
-    Real k = ((1.0 - 1.0 / n) * C - 1.0) / (C - 1.0);
-    if (k < 0.0)
-      k = 0.0;
-    else if (k > 1.0)
-      k = 1.0;
+      case 0: // Bazant
+      {
+        // adsorption heat of water vapor
+        Real adsorption_heat_water =
+            60 * 1000 / 0.018; // an average value (60 kJ/mol) estimated for different cement paste
+                               // by Poyet and Charles (2009)
+        Real density_water = 1000;       // kg/m3
+        Real heat_capacity_water = 4180; // J/(kg-C)
+        _ca[qp] = adsorption_heat_water;
+        _cw[qp] = density_water * heat_capacity_water;
+        _moisture_capacity[qp] = 1.0;
 
-    Real V_ct = 0.9;
-    switch (_cement_type)
-    {
-      case 0:
-        V_ct = 0.9;
-        break;
-      case 1:
-        V_ct = 1.0;
-        break;
-      case 2:
-        V_ct = 0.85;
-        break;
-      case 3:
-        V_ct = 0.6;
+        if (T <= 95.0)
+        {
+          Real alfa_T = 0.05 + 0.95 / 70 * (T - 25); // Bazant and Thonguthai 1979
+          Real f1_h =
+              alfa_T + (1 - alfa_T) / (1.0 + std::pow(((1.0 - H) / (1 - _Hc)),
+                                                      _n_power)); // Bazant and Thonguthai 1979
+          Real f2_T = std::exp(
+              2700.0 * (1.0 / (25.0 + 273.15) - 1.0 / (T + 273.15))); // Bazant and Thonguthai 1979
+          _Dh[qp] = _D1 * f1_h * f2_T;
+        }
+        else
+        {
+          Real f2_95 = std::exp(
+              2700.0 * (1.0 / (25.0 + 273.15) - 1.0 / (95 + 273.15))); // Bazant and Thonguthai 1979
+          Real f3_T =
+              std::exp((T - 95.0) / (0.881 + 0.214 * (T - 95.0))); // Bazant and Thonguthai 1979
+          _Dh[qp] = _D1 * f2_95 * f3_T;                            // Bazant and Thonguthai 1979
+        }
+        // compute the coupled mositure diffusivity due to thermal gradient
+        _Dht[qp] = _alfa_Dht * _Dh[qp];
         break;
-      default:
-        mooseError("Unknown cement type in mositure capacity calculations");
+      }
+
+      case 1: // Mensi (it is not meant for humidity transfer in concrete, however it is meant for
+              // drying of concrete (change in water content in concrete))
+      {
+        _ca[qp] = 0.;
+        _cw[qp] = 0.;
+        _moisture_capacity[qp] = 1.0;
+        Real C1 = H * _C0;
+        _Dh[qp] = _A * std::exp(_B * C1);
+        _Dht[qp] = 0.;
         break;
-    }
-
-    Real V_wc = 0.985;
-    if (_water_to_cement < 0.3)
-      V_wc = 0.985;
-    else if (_water_to_cement >= 0.3 && _water_to_cement <= 0.7)
-      V_wc = 0.85 + 0.45 * _water_to_cement;
-    else
-      V_wc = 1.165;
+      }
+
+      case 2: // Xi model
+      {
+        _ca[qp] = 0.;
+        _cw[qp] = 0.;
+        Real C = std::exp(855.0 / (T + 273.15));
+        Real N_ct = 1.1;
+        switch (_cement_type)
+        {
+          case 0:
```

Comment:
Please add comments regarding what each of these cases corresponds to.

Reply:
Done!